File: src/modules/sizing/FullAllocationSizing.cs
using System;
using System.Collections.Generic;
using StrategyFramework.Interfaces;

namespace StrategyFramework.Modules.Sizing
{
    /// <summary>
    /// Full allocation position sizing - uses 100% of net liquidation value
    /// </summary>
    public class FullAllocationSizing : IPositionSizingModule
    {
        public string Name => "FullAllocation_100%";
        public Dictionary<string, object> Parameters { get; set; }

        public FullAllocationSizing()
        {
            Parameters = new Dictionary<string, object>
            {
                { "AllocationPercent", 1.0m } // 100%
            };
        }

        public decimal CalculatePositionSize(SecurityData data, decimal portfolioValue, decimal availableCash, bool isLong)
        {
            // Use 99% of available cash to allow for rounding and fees
            var dollarAmount = availableCash * 0.99m;
            var shares = Math.Floor(dollarAmount / data.Close);
            
            return isLong ? shares : -shares;
        }
    }
}

File: modules/sizing/full_allocation_sizing.py
"""
Full allocation position sizing module
Allocates 100% of available capital to position
"""
from framework.interfaces import IPositionSizingModule
from typing import Dict, Any


class FullAllocationSizing(IPositionSizingModule):
    """Position sizing that uses 100% of available capital"""
    
    def __init__(self):
        self.module_name = "FullAllocationSizing"
    
    def calculate_size(self, data: Dict[str, Any]) -> float:
        """
        Calculate position size as 100% allocation
        Args:
            data: Dictionary containing account info (not used for full allocation)
        Returns:
            Position size as 1.0 (100% of portfolio)
        """
        return 1.0
    
    def get_module_name(self) -> str:
        """Return module name for identification"""
        return self.module_name